### PR TITLE
Add breaking change page for legacy-js-api

### PR DESCRIPTION
**Pull request by jathak**

### CODE REVIEW

File: source/documentation/breaking-changes/legacy-js-api.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Importers in the legacy API consisted of a single function that took in the
+dependency rule URL and the URL of the containing stylesheet (as well as a
+`done` callback for asynchronous importers) and return an object with either
+a `file` path on disk or the `contents` of the stylesheet to be loaded.
+
+Modern API [`Importer`]s instead contain two methods: `canonicalize`, which takes
+in a rule URL and returns the canonical form of that URL; and `load`, which
+takes in a canonical URL and returns an object with the contents
+of the loaded stylesheet. Asynchronous importers have both of these methods
```
Comment (nex3) [inline source/documentation/breaking-changes/legacy-js-api.md:36]: I'd add a sentence here explaining that this split ensures that the same module is only loaded once and that relative URLs work consistently.

File: source/documentation/breaking-changes/legacy-js-api.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+### Custom Functions
+
+In the legacy JS API, custom functions took a separate JS argument for each
+Sass argument, with an additional `done` callback for asynchronous custom
+functions. In the modern API, custom functions instead take a single JS argument
+that contains a list of all Sass arguments, with asynchronous custom functions
+returning a promise.
+
+The modern API also uses a much more robust [`Value`] class that supports
+type assertions and easy map and list lookups.
```
Comment (nex3) [inline source/documentation/breaking-changes/legacy-js-api.md:56]: I'd add "all Sass value types" to the list of things the new function API supports

File: source/documentation/breaking-changes/legacy-js-api.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+tool to tell it to use the modern API.
+
+Webpack should already use the modern API by default, but if you're getting
+warnings, set `api` to `"modern"` or `"modern-compiler"`.
+See [Webpack's documentation] for more details.
+
+Vite still defaults to the legacy API, but you can similarly switch it by
+setting `api` to `"modern"` or `"modern-compiler"`. See [Vite's documentation]
+for more details.
+
```
Comment (nex3) [inline source/documentation/breaking-changes/legacy-js-api.md:73]: Maybe also add a note that for other bundlers or frameworks, check the documentation or issue tracker for information about supporting the modern Sass API?

**Review by nex3 — CHANGES_REQUESTED** [review]

File: source/documentation/breaking-changes/legacy-js-api.md (hunk 1 above)
Comment (jathak) [reply source/documentation/breaking-changes/legacy-js-api.md:36]: Done

File: source/documentation/breaking-changes/legacy-js-api.md (hunk 2 above)
Comment (jathak) [reply source/documentation/breaking-changes/legacy-js-api.md:56]: Done

File: source/documentation/breaking-changes/legacy-js-api.md (hunk 3 above)
Comment (jathak) [reply source/documentation/breaking-changes/legacy-js-api.md:73]: Done

**Review by nex3 — APPROVED** [review]

**Pull request merged**